docs/diagrams/FindCommandSequenceDiagram.puml

```diff
@@ -3,32 +3,45 @@
 
 box "Logic" LOGIC_COLOR_T1
 
+Participant ":LogicManager" as logic LOGIC_COLOR
+Participant ":ReeveParser" as rp LOGIC_COLOR
+Participant ":FindCommandParser" as fcp LOGIC_COLOR
+Participant "d:FindStudentDescriptor" as fcd LOGIC_COLOR
 Participant "command:FindCommand" as fc LOGIC_COLOR
-Participant ":FindStudentDescriptor" as fcd LOGIC_COLOR
-Participant "result:CommandResult" as res LOGIC_COLOR
 
 
-[-> fc : execute(model)
-activate fc
+[-> logic : execute
+activate logic
+logic -> rp ++: parseCommand(commandText)
+create fcp
+rp -> fcp
+rp -> fcp ++: parse(args)
+create fcd
+fcp -> fcd ++:
+return d
+
+create fc
+fcp -> fc ++: new FindCommand(d)
+return command
+return command
+return command
+destroy fcp
+
+logic -> fc ++: execute(model)
 fc -> fcd++: getPredicates()
-fcd -> fcd: getNamePredicate()
-fcd -> fcd: getSchoolPredicate()
-fcd -> fcd: getYearPredicate()
 return consolidatedPredicates
 
-
 end box
 
 box "Model" MODEL_COLOR_T1
-Participant "model:Model" as model MODEL_COLOR
 
-fc -> model: updateFilteredPersonList(consolidatedPredicates)
+Participant ":Model" as model MODEL_COLOR
 
-fc -> model: getFilteredPersonList()
-create res
-fc -> res ++
-return result
+fc -> model++: updateFilteredPersonList(consolidatedPredicates)
```

Comment:
Perhaps you mean `updateFilteredStudentList(...)`? Maybe `predicates` would be better than `consolidatedPredicates` for brevity.